### PR TITLE
fix for sync algorithm with live dependencies of non-live extensions

### PR DESCRIPTION
**Pull request by czue**
@proteusvacuum probably good to confirm this seems right

http://manage.dimagi.com/default.asp?243839

probably easier commit-wise though a couple tests that got added and later removed...

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -609,31 +613,24 @@ def __repr__(self):
...
     @staticmethod
-    def get_all_dependencies(case_id, child_index_tree, extension_index_tree, closed_cases=None,
+    def get_all_dependencies(case_id, child_index_tree, extension_index_tree,
                              cached_child_map=None, cached_extension_map=None):
-        """Takes a child and extension index tree and returns returns a set of all dependencies of <case_id>
+        """
+        Takes a child and extension index tree and returns returns a set of all dependencies of <case_id>
 
-        Traverse each incoming index, return each touched case. Stop traversing
-        incoming extensions if they lead to closed cases.
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: I'm not sure why this condition was here in the first place. @proteusvacuum do you know?

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/ex-submodules/casexml/apps/phone/models.py, line 836 - E121 continuation line under-indented for hanging indent
* corehq/ex-submodules/casexml/apps/phone/models.py, line 977 - E501 line too long (119 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/models.py, line 1087 - E501 line too long (116 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/models.py, line 1104 - E501 line too long (118 > 115 characters)
* corehq/ex-submodules/casexml/apps/phone/tests/test_index_tree.py, line 449 - E501 line too long (118 > 115 characters)

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -835,34 +811,44 @@ def _get_relevant_cases(self, case_id):
...
         Mark all relevant cases that are open and have no outgoing extension indexes
-        as available. Traverse incoming extension indexes which don't lead to closed
+        as available.
+
+        Traverse incoming extension indexes which don't lead to closed
         cases, mark all touched cases as available
         """
+        def _case_is_available(case):
+            return (
+                case not in self.dependent_case_ids_on_phone  # all live cases are available
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: this seemed logical to me, though not sure i'm overlooking something funky with extensions

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 3]
```diff
@@ -21,7 +21,11 @@
 from casexml.apps.phone.checksum import Checksum, CaseStateHash
 import logging
 
-logger = logging.getLogger('phone.models')
+
+def _get_logger():
+    # for some strange reason if you define logger the normal way it gets silenced in tests.
+    # this is a hacky workaround to that.
+    return logging.getLogger(__name__)
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: I spent what felt like a reasonable amount of time trying to find a reason why this was happening and eventually gave up.

**Comment by czue** [comment]
@proteusvacuum ping me on slack before you review this. i'm confused what the behavior is supposed to be if you _own_ a closed extension of a closed host. the failing test seems to indicate that it should also be purged. @ctsims might also need you to weigh in on this.

**Comment by czue** [comment]
> i'm confused what the behavior is supposed to be if you _own_ a closed extension of a closed host. the failing test seems to indicate that it should also be purged. @ctsims might also need you to weigh in on this.

the mobile tests seem to validate this is right so going to tweak HQ's behavior accordingly.

**Pull request closed without merging**